app/services/maat_link_creator.rb

```diff
@@ -65,18 +66,6 @@ def post_laa_reference_to_common_platform(offence)
     raise StandardError, "Error posting LAA Reference to Common Platform" unless response.success?
   end
 
-  def persist_laa_reference
-    existing_laa_ref = LaaReference.find_by(
-      defendant_id: laa_reference.defendant_id,
-      maat_reference: laa_reference.maat_reference,
-      linked: true,
-    )
```

Comment:
This method is the main culprit for the bug.
I've refactored it. It's now called `adjust_link_and_save!`